sway-lib-std/src/vec.sw

```diff
@@ -248,3 +249,27 @@ impl<T> Vec<T> {
         index_ptr.write(value);
     }
 }
+
+impl<T> AsSlice for Vec<T> {
+    /// Returns a slice to all of the elements in the vector.
+    fn as_slice(self) -> raw_slice {
+        raw_slice::from_parts::<T>(self.buf.ptr(), self.len)
+    }
+}
+
+impl<T> From<raw_slice> for Vec<T> {
+    fn from(slice: raw_slice) -> Vec<T> {
+        let buf = RawVec {
+            ptr: slice.ptr(),
+            cap: slice.len::<T>(),
+        };
+        Self {
+            buf,
+            len: buf.cap,
+        }
+    }
+
+    fn into(self) -> raw_slice {
+        asm(ptr: (self.buf.ptr(), self.len)) { ptr: raw_slice }
+    }
+}
```

Comment:
I like all of this.

Reply:
I also like that `Vec` doesn't have to change. It's nice to keep it somewhat in sync with how Rust's `Vec` looks like.

Reply:
Also this is a non-breaking change so that's a bonus :D

Reply:
Yeah I was so happy to be able to avoid that :D